Nos/Views/NoteComposer/NoteComposer.swift

```diff
@@ -321,18 +370,79 @@ struct NoteComposer: View {
             Log.error("Error when posting: \(error.localizedDescription)")
         }
     }
+
+    /// Builds a JSONEvent object for a given text and key pair.
+    /// - Parameters:
+    ///   - attributedText: The text the user wrote.
+    ///   - keyPair: Key pair of the logged in user.
+    /// - Returns: A JSONEvent object suitable to be published to Nostr.
+    private func buildJSONEvent(attributedText: AttributedString, keyPair: KeyPair) -> JSONEvent {
```

Comment:
Now that you're refactored this out of `publishPost`, it's easier for me to see that it belongs somewhere else. I understand that it's only used by `NoteComposer`, but to me that doesn't mean it needs to live inside of `NoteComposer`. I'd suggest a new type in a new file, but the name is the hard part here... `JSONBuilder`? `JSONEventBuilder`? Neither of those seems great. An `init` on `JSONEvent`, possibly even in an extension? That's what we're actually doing -- initializing a JSONEvent from a handful of values, right? I'm starting to think that might be the right way to go. Something like:

```
init(
    attributedText: AttributedText,
    keyPair: KeyPair,
    expirationTime: ...,
    replyToNote: ...
)
```

Reply:
It's ready to be reviewed again. I added one comment pointing to the line that was causing an issue and preventing me from asking for a re-review.